### PR TITLE
Add quotecall on max scenario for swap() and cometSupply()

### PR DESCRIPTION
**Pull request by cwang25**

### CODE REVIEW

File: src/builder/QuarkBuilder.sol [hunk 1]
```diff
@@ -402,6 +402,14 @@ contract QuarkBuilder {
             chainAccountsList = Accounts.findChainAccountsWithPaymentInfo(chainAccountsList, payment);
         }
 
+        // Initialize comet supply max flag
+        bool isMaxSupply = cometSupplyIntent.amount == type(uint256).max;
+        // Convert transferIntent to user aggregated balance
```
Comment (kevincheng96) [inline src/builder/QuarkBuilder.sol:407]: Comment needs to be updated. Same for the other function

File: src/builder/QuarkBuilder.sol [hunk 2]
```diff
@@ -836,10 +844,18 @@ contract QuarkBuilder {
...
+        // Initialize swap max flag (when sell amount is max)
+        bool isMaxSwap = swapIntent.sellAmount == type(uint256).max;
+        // Convert transferIntent to user aggregated balance
+        if (isMaxSwap) {
+            swapIntent.sellAmount = Accounts.totalAvailableAsset(sellAssetSymbol, chainAccountsList, payment);
+        }
+
         assertFundsAvailable(swapIntent.chainId, sellAssetSymbol, swapIntent.sellAmount, chainAccountsList, payment);
 
         // TODO: When should we use quotecall?
```
Comment (kevincheng96) [inline src/builder/QuarkBuilder.sol:857]: Can remove this comment now

File: src/builder/QuarkBuilder.sol [hunk 3]
```diff
@@ -410,7 +418,7 @@ contract QuarkBuilder {
             payment
         );
 
-        bool useQuotecall = false; // never use Quotecall
+        bool useQuotecall = isMaxSupply; // never use Quotecall
```
Comment (kevincheng96) [inline src/builder/QuarkBuilder.sol:421]: Can remove comment

File: test/builder/QuarkBuilderCometSupply.t.sol [hunk 4]
```diff
@@ -150,6 +150,95 @@ contract QuarkBuilderCometSupplyTest is Test, QuarkBuilderTest {
...
+            quarkStates: quarkStates_(address(0xb0b), 2),
+            assetPositionsList: assetPositionsList_(8453, address(0xb0b), uint256(0)),
+            cometPositions: emptyCometPositions_()
+        });
+        chainAccountsList[2] = Accounts.ChainAccounts({
+            chainId: 7777,
+            quarkStates: quarkStates_(address(0xc0b), 5),
+            assetPositionsList: assetPositionsList_(7777, address(0xc0b), uint256(0)),
+            cometPositions: emptyCometPositions_()
+        });
```
Comment (kevincheng96) [inline test/builder/QuarkBuilderCometSupply.t.sol:173]: You don't need all this right?

File: test/builder/QuarkBuilderCometSupply.t.sol [hunk 5]
```diff
@@ -444,6 +533,141 @@ contract QuarkBuilderCometSupplyTest is Test, QuarkBuilderTest {
         assertNotEq(result.eip712Data.hashStruct, hex"", "non-empty hashStruct");
     }
 
+    function testCometSupplyMaxWithBridge() public {
+        QuarkBuilder builder = new QuarkBuilder();
+        QuarkBuilder.BuilderResult memory result = builder.cometSupply(
+            cometSupply_(8453, type(uint256).max),
+            chainAccountsList_(6e6), // holding 3 USDC in total across chains 1, 8453
+            paymentUsd_()
```
Comment (kevincheng96) [inline test/builder/QuarkBuilderCometSupply.t.sol:541]: Can you test with token payment instead? To make sure that Quotecall is being used here

File: test/builder/QuarkBuilderSwap.t.sol [hunk 6]
```diff
@@ -396,6 +396,101 @@ contract QuarkBuilderSwapTest is Test, QuarkBuilderTest {
...
+        chainAccountsList[2] = Accounts.ChainAccounts({
+            chainId: 7777,
+            quarkStates: quarkStates_(address(0xc0b), 5),
+            assetPositionsList: assetPositionsList_(7777, address(0xc0b), uint256(0)),
+            cometPositions: emptyCometPositions_()
+        });
+
+        QuarkBuilder.BuilderResult memory result = builder.swap(
+            buyWeth_(1, usdc_(1), type(uint256).max, 3e18, address(0xa11ce), BLOCK_TIMESTAMP), // swap max on chain 1
+            chainAccountsList, // holding 3050 USDC in total chains 1
```
Comment (kevincheng96) [inline test/builder/QuarkBuilderSwap.t.sol:425]: Where is 3050 coming from?

File: test/builder/QuarkBuilderCometSupply.t.sol (hunk 4 above)
Comment (cwang25) [reply test/builder/QuarkBuilderCometSupply.t.sol:173]: Let me try to remove those,  but I recalled if the list is not match with some other inputs, some error occurs. (forgot what error it got hit),  so I kept it as close as the one we've been used in setup function, but modify it to be 0 in other chain.)

File: src/builder/QuarkBuilder.sol (hunk 1 above)
Comment (cwang25) [reply src/builder/QuarkBuilder.sol:407]: oops yeah, will update those comments.

File: test/builder/QuarkBuilderSwap.t.sol (hunk 6 above)
Comment (cwang25) [reply test/builder/QuarkBuilderSwap.t.sol:425]: outdated comment :P
Was testing with 3050 originally, then went with 9005 but forgot to chant this commetn.

File: src/builder/QuarkBuilder.sol (hunk 3 above)
Comment (cwang25) [reply src/builder/QuarkBuilder.sol]: removed 👍

File: test/builder/QuarkBuilderCometSupply.t.sol (hunk 4 above)
Comment (cwang25) [reply test/builder/QuarkBuilderCometSupply.t.sol:173]: ok looks like this one worked out well haha.  Will removed other acount balnce.

File: test/builder/QuarkBuilderCometSupply.t.sol (hunk 5 above)
Comment (cwang25) [reply test/builder/QuarkBuilderCometSupply.t.sol:541]: No problem 👍 
I have create anotherone to test with token payment.

**Review by kevincheng96 — APPROVED** [review]

**Pull request merged**